12-offer-encoding.md

```diff
@@ -0,0 +1,846 @@
+    2. data:
+        * [`tu64`:`msat`]
+    1. type: 172 (`invoice_fallbacks`)
+    2. data:
+        * [`...*fallback_address`:`fallbacks`]
+    1. type: 174 (`invoice_features`)
+    2. data:
+        * [`...*byte`:`features`]
+    1. type: 176 (`invoice_code`)
+    2. data:
+        * [`...*utf8`:`code`]
+    1. type: 240 (`signature`)
+    2. data:
+        * [`bip340sig`:`sig`]
+
+1. subtype: `blinded_payinfo`
+2. data:
+   * [`u32`:`fee_base_msat`]
+   * [`u32`:`fee_proportional_millionths`]
+   * [`u16`:`cltv_expiry_delta`]
+   * [`u64`:`htlc_minimum_msat`]
+   * [`u64`:`htlc_maximum_msat`]
+   * [`u16`:`flen`]
+   * [`flen*byte`:`features`]
+
+1. subtype: `fallback_address`
+2. data:
+   * [`byte`:`version`]
+   * [`u16`:`len`]
+   * [`len*byte`:`address`]
+
+## Invoice Features
+
+| Bits | Description                      | Name           |
+|------|----------------------------------|----------------|
+| 16   | Multi-part-payment support       | MPP/compulsory |
+| 17   | Multi-part-payment support       | MPP/optional   |
+
+The 'MPP support' invoice feature indicates that the payer MUST (16) or
+MAY (17) use multiple part payments to pay the invoice.
+
+Some implementations may not support MPP (e.g. for small payments), or
+may (due to capacity limits on a single channel) require it.
+
+## Requirements
+
+A writer of an invoice:
+  - MUST set `invoice_created_at` to the number of seconds since Midnight 1
+    January 1970, UTC when the offer was created.
+  - MUST set `invoice_amount` to the minimum amount it will accept, in units of 
+    the minimal lightning-payable unit (e.g. milli-satoshis for bitcoin) for
+    `invreq_chain`.
+  - if the invoice is in response to an `invoice_request`:
+    - MUST copy all non-signature fields from the `invoice_request` (including unknown fields).
+    - if `invreq_amount` is present:
+      - MUST set `invoice_amount` to `invreq_amount`
+    - otherwise:
+      - MUST set `invoice_amount` to the *expected amount*.
+  - otherwise (invoice not requested, e.g. for user to scan directly):
+    - MUST set `invreq_chain` as it would for an invoice_request.
+    - MUST set `offer_description` and `offer_node_id` as it would for an offer.
+    - MUST NOT set `invreq_payer_id`.
+  - MUST set `invoice_payment_hash` to the SHA256 hash of the
+    `payment_preimage` that will be given in return for payment.
+  - if `offer_node_id` is not present:
+    - if the `invoice_request` was not communicated securely (i.e. privately):
+      - SHOULD set `invoice_code` to a short string for verification.
+    - MUST NOT set `signature`.
+  - otherwise:
+    - MUST NOT set `invoice_code`.
+    - MUST specify exactly one signature TLV element: `signature`.
+      - MUST set `sig` to the signature using `offer_node_id` as described in [Signature Calculation](#signature-calculation).
+  - if it requires multiple parts to pay the invoice:
+    - MUST set `invoice_features`.`features` bit `MPP/compulsory`
```

Comment:
Seems odd to say "requires multiple parts" given the payment amount may be small enough where MPP wouldn't be suitable. Is the intention to force multiple paths in order to enhance privacy?

Reply:
Presumably it requires multiple parts for capacity reasons...